plugins/modules/sda_extranet_policies_workflow_manager.py

```diff
@@ -0,0 +1,1027 @@
+            It calls the `get_extranet_policies` method to retrieve policy details. If the policy is found, it sets
+            `extranet_policy_exists` to `True` and extracts the policy ID and details. The method returns a tuple containing
+            the existence status, policy ID, and policy details.
+        """
+        # Initialize variables to default values
+        extranet_policy_exists = False
+        extranet_policy_id = None
+
+        extranet_policy_name = config.get("extranet_policy_name")
+        extranet_policy_details = self.get_extranet_policies(extranet_policy_name)
+
+        # Check if the policy details were retrieved successfully
+        if extranet_policy_details:
+            extranet_policy_exists = True
+            extranet_policy_id = extranet_policy_details["id"]
+
+        return (extranet_policy_exists, extranet_policy_id, extranet_policy_details)
+
+    def compare_extranet_policies(self, extranet_policy_details, update_extranet_policy_params):
+        """
+        Compare the details of two extranet policies to check if they are equivalent.
+        Parameters:
+            - extranet_policy_details (dict): A dictionary containing the current details of the extranet policy.
+            - update_extranet_policy_params (dict): A dictionary containing the updated policy parameters to compare against.
+        Returns:
+            bool: `True` if all values for the keys match between the two dictionaries, `False` otherwise.
+        Description:
+            This method compares the details of two extranet policies by iterating over each key in the `extranet_policy_details`
+            dictionary and checking if the corresponding values in the `update_extranet_policy_params` dictionary match.
+            Lists are compared regardless of order, while other values are compared directly. The method returns `True` if
+            all values are equivalent, and `False` if any values differ.
+        """
+        # Iterate over each key in the extranet policy details and compare the details
+        for key in extranet_policy_details:
+            value1 = extranet_policy_details.get(key)
+            value2 = update_extranet_policy_params.get(key)
+
+            self.log("Comparing key: {0},  existing_value: {1}, requested_value: {2}".format(key, value1, value2))
+
+            if key == "fabricIds":
+                if value1 and not value2:
+                    continue
+
+            if isinstance(value1, list) and isinstance(value2, list):
+                # Compare lists regardless of order
+                if sorted(value1) != sorted(value2):
+                    return False
+            else:
+                # Compare values directly
+                if value1 != value2:
+                    return False
+
+        return True
+
+    def add_extranet_policy(self, add_extranet_policy_params):
```

Comment:
I cannot change the names because I have automated the get_diff_merged to execute respective functions based on the parameters set - add or update

    def get_diff_merged(self):
        """
        Executes actions based on the desired state parameters and checks their status.
        Parameters:
            - None
        Returns:
            self: The instance of the class, allowing for method chaining.
        Description:
            This method iterates through a map of action parameters to their corresponding functions for execution and status
            checking. For each action parameter present in the desired state (`want`), the associated action function is called
            to perform the action, and the corresponding status function is used to check the result. It ensures that all actions
            specified in the desired state are executed and their statuses are verified. The method returns the instance for method
            chaining.
        """
        action_map = {
            "add_extranet_policy_params": (self.add_extranet_policy, self.get_add_extranet_policy_status),
            "update_extranet_policy_params": (self.update_extranet_policy, self.get_update_extranet_policy_status),
        }

        for action_param, (action_func, status_func) in action_map.items():
            # Execute the action and check its status
            if self.want.get(action_param):
                result_task_id = action_func(self.want.get(action_param))
                status_func(result_task_id).check_return_status()
        return self